src/main/java/org/broadinstitute/hellbender/tools/spark/sv/AssemblyAlignmentParser.java

```diff
@@ -251,47 +247,63 @@
-                    }
-                    result.add(new CigarElement(element.getLength()+cigarElements.get(indexForFirstMaybeInsertion+1).getLength(), CigarOperator.S));
-                    if (numHardClippingBasesFromOneEnd!=0) { // some hard clipping happened
-                        result.add(cigarElements.get(cigarElements.size() - 1));
-                    }
-                }
-                return result;
-            } else {
-                return cigarElements;
-            }
+            cigarElements.set(indexOfFirstNonClippingOperation, new CigarElement(element.getLength(), CigarOperator.S));
+
+            return compactifyNeighboringSoftClippings(cigarElements);
+
+//            if (numSoftClippingBasesFromOneEnd!=0) { // compactify with possible neighboring clipping
+//                final List<CigarElement> result = new ArrayList<>(cigarElements.size());
+//                if (fromStart) {
+//                    if (numHardClippingBasesFromOneEnd!=0) { // some hard clipping happened
+//                        result.add(0, cigarElements.get(0));
+//                    }
+//                    result.add(new CigarElement(element.getLength()+cigarElements.get(indexOfFirstNonClippingOperation-1).getLength(), CigarOperator.S));
+//                    if (indexOfFirstNonClippingOperation!=cigarElements.size()-1) { // copy the rest, if any
+//                        result.addAll(cigarElements.subList(indexOfFirstNonClippingOperation+1, cigarElements.size()));
+//                    }
+//                } else {
+//                    if (indexOfFirstNonClippingOperation!=0) { // copy the rest, if any
+//                        result.addAll(cigarElements.subList(0, indexOfFirstNonClippingOperation));
+//                    }
+//                    result.add(new CigarElement(element.getLength()+cigarElements.get(indexOfFirstNonClippingOperation+1).getLength(), CigarOperator.S));
+//                    if (numHardClippingBasesFromOneEnd!=0) { // some hard clipping happened
+//                        result.add(cigarElements.get(cigarElements.size() - 1));
+//                    }
+//                }
+//                return result;
+//            } else {
+//                return cigarElements;
+//            }
         } else {
             return cigarElements;
         }
     }
 
+    @VisibleForTesting
+    static List<CigarElement> compactifyNeighboringSoftClippings(final List<CigarElement> cigarElements) {
+        final List<CigarElement> result = new ArrayList<>(cigarElements.size());
+        for (final CigarElement element : cigarElements) {
+            if (result.isEmpty() || result.get(result.size()-1).getOperator()!=element.getOperator()) {
+                result.add(element);
+            } else { // no additional checking because this is possible only for two neighboring soft clippings, one of which is converted from an insertion operation
```

Comment:
Are you sure about that (that this will never happen except for in the case of a converted soft clip)? We could be a little more conservative here and only do the merge if the adjacent elements are in fact soft clips. I just worry that there's some BWA edge case that could give two adjacent "M" operators or something, and this would squish them incorrectly into a new "S" element. Also the name of the method is specific to soft clips.